src/ResponseBuilder.php

```diff
@@ -35,7 +41,11 @@ private function getJsonDecodedResponse(string $body)
 
         $error = json_last_error();
         if (JSON_ERROR_NONE !== $error) {
-            throw new \UnexpectedValueException('Invalid JSON response.');
+            $message = 'Invalid JSON response.';
+            if (json_last_error_msg()) {
+                $message .= ' ' . json_last_error_msg();
```

Comment:
I think if we retrieve the value and then check it against false it would read better. I think ideally we should achieve something like:"Invalid JSON response: this is my json error", and maybe something like "unknown error" in the case of json_last_error_msg returns false. Wouldn't you agree?